### ISSUE TITLE
Updating auth to add google signin results in error

### ISSUE THREAD

**Issue by samlaf** [aws-amplify/amplify-cli#8280]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm i -g @aws-amplify/cli@6.0.1

### If applicable, what version of Node.js are you using?

v14.17.4

### Amplify CLI Version

6.0.1

### What operating system are you using?

ubuntu 20.04

### Amplify Categories

auth

### Amplify Commands

update

### Describe the bug

`amplify update auth` and adding google social signin results in the following error:
```
Successfully updated AdminQueriesf34b0e43 function locally
Successfully updated AdminQueries API locally
Error: Unable to update lambda function
    at AmplifyToolkit.updateTrigger [as _updateTrigger] (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/src/extensions/amplify-helpers/trigger-flow.ts:199:11)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at handleTriggers (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/src/provider-utils/awscloudformation/utils/trigger-flow-auth-helper.js:72:31)
    at lambdaTriggers (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts:170:46)
    at /home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts:65:3
    at /home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/src/provider-utils/awscloudformation/handlers/resource-handlers.ts:47:5
    at Object.executeAmplifyCommand (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/src/index.js:375:3)
    at executePluginModuleCommand (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:177:3)
    at Object.executeCommand (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/src/execution-manager.ts:29:5)
    at Object.run (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/src/index.ts:205:5)
There was an error updating the auth resource
Successfully updated resource auth locally
```

### Expected behavior

I expected the auth to get updated normally.

### Reproduction steps

My current backend looks like
![image](https://user-images.githubusercontent.com/9342524/134792808-b5fa2795-afdb-45aa-a4cf-8cebae00f607.png)

Let me know if any config file could be of use.

### Log output

<details>

```
2021-09-26T03:17:58.293Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["default"])
2021-09-26T03:17:58.294Z|info : amplify-provider-awscloudformation.system-config-manager.getProfiledAwsConfig.profileConfig([{"region":"us-east-1"}])
2021-09-26T03:17:58.294Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileCredentials(["default"])
2021-09-26T03:17:58.303Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["default"])
2021-09-26T03:17:58.303Z|info : amplify-provider-awscloudformation.system-config-manager.getProfiledAwsConfig.profileConfig([{"region":"us-east-1"}])
2021-09-26T03:17:58.304Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileCredentials(["default"])
2021-09-26T03:17:58.307Z|info : amplify-provider-awscloudformation.zip-util.downloadZip.s3.getFile([{"Key":"[***]rent-[***]ud-[***]d.zip"},null])
2021-09-26T03:17:58.310Z|info : amplify-provider-awscloudformation.aws-s3.s3.getFile([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]ment"}])
2021-09-26T03:17:59.475Z|info : amplify-provider-awscloudformation.aws-s3.getAllObjectKey.s3.listObjectVersions([{"Bucket":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]ment","Prefix":"hooks/"}])
2021-09-26T03:17:59.955Z|info : amplify-provider-awscloudformation.initialize-env.run.cfn.updateamplifyMetaFileWithStackOutputs([{"StackName":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612"}])
2021-09-26T03:17:59.961Z|info : amplify-provider-awscloudformation.aws-cfn.updateamplifyMetaFileWithStackOutputs.cfn.describeStackResources([{"StackName":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612"}])
2021-09-26T03:18:00.450Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]ueries-[***]IN0EVR/[***]5010-[***]8a-[***]eb-[***]3d-[***]f9bdf"}])
2021-09-26T03:18:00.452Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]appapi-[***]BVNWG/[***]5480-[***]8a-[***]eb-[***]ed-[***]0620f"}])
2021-09-26T03:18:00.452Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]auth-[***]ZKZX74/[***]9530-[***]8a-[***]eb-[***]c3-[***]8d317"}])
2021-09-26T03:18:00.453Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]olGroups-[***]L3MDOF/[***]8800-[***]8a-[***]eb-[***]fe-[***]69e4d"}])
2021-09-26T03:18:00.454Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]riesf34b0e43-[***]TKO0JE/[***]1720-[***]8a-[***]eb-[***]10-[***]50fd5"}])
2021-09-26T03:18:00.455Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]upsToToken-[***]FF2QO/[***]fe60-[***]8b-[***]eb-[***]09-[***]1bf57"}])
2021-09-26T03:18:00.456Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]Confirmation-[***]O8JJC/[***]8aa0-[***]8a-[***]eb-[***]92-[***]6a645"}])
2021-09-26T03:18:00.456Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]enGeneration-[***]P6BY3D/[***]08a0-[***]8a-[***]eb-[***]79-[***]9cfa9"}])
2021-09-26T03:18:00.457Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]media-[***]4NB4LZ/[***]d040-[***]8a-[***]eb-[***]85-[***]7c4a3"}])
2021-09-26T03:18:00.920Z|info : amplify-provider-awscloudformation.aws-s3.s3.getFile([{"Key":"[***]ify-[***]json","Bucket":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]ment"}])
2021-09-26T03:18:09.356Z|info : amplify pull core  
2021-09-26T03:18:09.399Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-26T03:18:15.762Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["default"])
2021-09-26T03:18:15.763Z|info : amplify-provider-awscloudformation.system-config-manager.getProfiledAwsConfig.profileConfig([{"region":"us-east-1"}])
2021-09-26T03:18:15.763Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileCredentials(["default"])
2021-09-26T03:18:15.769Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileConfig(["default"])
2021-09-26T03:18:15.769Z|info : amplify-provider-awscloudformation.system-config-manager.getProfiledAwsConfig.profileConfig([{"region":"us-east-1"}])
2021-09-26T03:18:15.769Z|info : amplify-provider-awscloudformation.system-config-manager.getProfileCredentials(["default"])
2021-09-26T03:18:15.773Z|info : amplify-provider-awscloudformation.zip-util.downloadZip.s3.getFile([{"Key":"[***]rent-[***]ud-[***]d.zip"},null])
2021-09-26T03:18:15.776Z|info : amplify-provider-awscloudformation.aws-s3.s3.getFile([{"Key":"[***]rent-[***]ud-[***]d.zip","Bucket":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]ment"}])
2021-09-26T03:18:17.066Z|info : amplify-provider-awscloudformation.aws-s3.getAllObjectKey.s3.listObjectVersions([{"Bucket":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]ment","Prefix":"hooks/"}])
2021-09-26T03:18:17.302Z|info : amplify-provider-awscloudformation.initialize-env.run.cfn.updateamplifyMetaFileWithStackOutputs([{"StackName":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612"}])
2021-09-26T03:18:17.306Z|info : amplify-provider-awscloudformation.aws-cfn.updateamplifyMetaFileWithStackOutputs.cfn.describeStackResources([{"StackName":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612"}])
2021-09-26T03:18:17.693Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]ueries-[***]IN0EVR/[***]5010-[***]8a-[***]eb-[***]3d-[***]f9bdf"}])
2021-09-26T03:18:17.694Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]appapi-[***]BVNWG/[***]5480-[***]8a-[***]eb-[***]ed-[***]0620f"}])
2021-09-26T03:18:17.695Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]auth-[***]ZKZX74/[***]9530-[***]8a-[***]eb-[***]c3-[***]8d317"}])
2021-09-26T03:18:17.696Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]olGroups-[***]L3MDOF/[***]8800-[***]8a-[***]eb-[***]fe-[***]69e4d"}])
2021-09-26T03:18:17.697Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]riesf34b0e43-[***]TKO0JE/[***]1720-[***]8a-[***]eb-[***]10-[***]50fd5"}])
2021-09-26T03:18:17.698Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]upsToToken-[***]FF2QO/[***]fe60-[***]8b-[***]eb-[***]09-[***]1bf57"}])
2021-09-26T03:18:17.699Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]Confirmation-[***]O8JJC/[***]8aa0-[***]8a-[***]eb-[***]92-[***]6a645"}])
2021-09-26T03:18:17.700Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]enGeneration-[***]P6BY3D/[***]08a0-[***]8a-[***]eb-[***]79-[***]9cfa9"}])
2021-09-26T03:18:17.700Z|info : amplify-provider-awscloudformation.aws-cfn.describeStack.cfn.describeStacks([{"StackName":"[***]rmation:us-[***]st-[***]17:stack/[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]media-[***]4NB4LZ/[***]d040-[***]8a-[***]eb-[***]85-[***]7c4a3"}])
2021-09-26T03:18:18.120Z|info : amplify-provider-awscloudformation.aws-s3.s3.getFile([{"Key":"[***]ify-[***]json","Bucket":"[***]ify-[***]eb-[***]pp-[***]ing-[***]612-[***]ment"}])
2021-09-26T03:19:30.906Z|info : amplify update auth  
2021-09-26T03:19:30.948Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-26T03:19:50.725Z|info : amplify update auth  
2021-09-26T03:19:50.766Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-26T03:21:26.101Z|info : amplify update auth  
2021-09-26T03:21:26.144Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-26T03:31:49.350Z|info : amplify version core  {"version":true,"yes":false}
2021-09-26T03:31:55.282Z|info : amplify update auth  
2021-09-26T03:31:55.327Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-26T03:34:14.325Z|info : amplify version core  
2021-09-26T03:41:03.470Z|info : amplify status core  {"yes":false}
2021-09-26T03:41:03.514Z|info : amplify-cli-core.banner-message/index.ts.fetch banner messages from https://aws-amplify.github.io/amplify-cli/banner-message.json({}
2021-09-26T03:41:07.840Z|info : amplify-provider-awscloudformation.display-helpful-urls.showCognitoSandBoxMessage(["auth"])


```

</details>

**Comment by josefaidt** [comment]
Hey @samlaf :wave: thanks for raising this! I stepped through the reproduction with Admin Queries API and a Post Confirmation trigger (add user to group) enabled, but was unable to reproduce the error you're experiencing. I have a few follow-up questions:

- in the screenshot, I notice the `auth` resource itself does not show an update. Were changes performed to the two resources shown as "update" before attempting `amplify auth update`? Or was this captured after the failed update attempt?
- what type of Post Confirmation trigger is in use and does it have any customizations applied beyond what scaffolding the CLI provides?

And I have followed these reproduction steps:

1. `amplify init -y`
2. `amplify add auth`, manual configuration, add groups (`users` and `admins`), add Admin Queries API restricted to `admins`, and added the `add user to group` Post Confirmation Cognito trigger to add new members to `users`
3. `amplify push -y`
4. `amplify update auth`, manual configuration, enable OAuth, add Google with fake credentials
   ```
    Using service: Cognito, provided by: awscloudformation
    What do you want to do? Walkthrough all the auth configurations
    Select the authentication/authorization services that you want to use: User Sign-Up, Sign-In, connected with AWS IAM controls (E
    nables per-user Storage features for images or other content, Analytics, and more)
    Allow unauthenticated logins? (Provides scoped down permissions that you can control via AWS IAM) No
    Do you want to enable 3rd party authentication providers in your identity pool? No
    Do you want to add User Pool Groups? Yes
    ? Select any user pool groups you want to delete: 
    ? Do you want to add another User Pool Group No
    ✔ Sort the user pool groups in order of preference · admins, users
    Do you want to add an admin queries API? Yes
    ? Do you want to restrict access to the admin queries API to a specific Group Yes
    ? Select the group to restrict access with: admins
    Multifactor authentication (MFA) user login options: OFF
    Email based user registration/forgot password: Enabled (Requires per-user email entry at registration)
    Please specify an email verification subject: Your verification code
    Please specify an email verification message: Your verification code is {####}
    Do you want to override the default password policy for this User Pool? No
    Specify the app's refresh token expiration period (in days): 30
    Do you want to specify the user attributes this app can read and write? No
    Do you want to enable any of the following capabilities? Add User to Group
    Do you want to use an OAuth flow? Yes
    What domain name prefix do you want to use? 828091eb5ca6-91eb5ca6
    Enter your redirect signin URI: http://localhost/signin/
    ? Do you want to add another redirect signin URI No
    Enter your redirect signout URI: http://localhost/signout/
    ? Do you want to add another redirect signout URI No
    Select the OAuth flows enabled for this project. Authorization code grant
    Select the OAuth scopes enabled for this project. Phone, Email, OpenID, Profile, aws.cognito.signin.user.admin
    Select the identity providers you want to configure for your user pool: Google
      
    You've opted to allow users to authenticate via Google.  If you haven't already, you'll need to go to https://developers.google.
    com/identity and create an App ID. 
    
    Enter your Google Web Client ID for your OAuth flow:  fakeclientid
    Enter your Google Web Client Secret for your OAuth flow:  fakeclientsecret
    ? Do you want to configure Lambda Triggers for Cognito? Yes
    ? Which triggers do you want to enable for Cognito Post Confirmation
    ? What functionality do you want to use for Post Confirmation Add User To Group
    Successfully updated AdminQueries760811c0 function locally
    Successfully updated AdminQueries API locally
    Successfully updated the Lambda function locally
    Successfully updated auth resource 828027d8391b27d8391b locally
   ```
5. Observe no error, `amplify push -y` then succeeds

**Comment by samlaf** [comment]
Hi @josefaidt, thanks for the prompt reply!
1. No updates were performed before `amplify update auth`. This was captured after the failed attempt.
![image](https://user-images.githubusercontent.com/9342524/135115446-524f166e-d2c7-48bc-86e8-813a30a9eb6d.png)
This is before doing anything. The API showing update is only because of syntactic reasons:
![image](https://user-images.githubusercontent.com/9342524/135115837-7bc0100f-98a2-4f98-8c23-c341b33ba352.png)
![image](https://user-images.githubusercontent.com/9342524/135115864-78fdbbd9-34ce-47e0-98c7-3a17f5da935f.png)
I tried both `amplify pull` and not before `amplify update auth` and both fail with the same error, which is what I sent in the original post.

2. the amplify setup I am working with was built by a colleague of mine more than a year ago, and he doesn't remember the details. Does the amplify CLI provide an initial `authPostConfirmation` function? I see that ours was changed, and the code is:
```
/* eslint-disable-line */ const aws = require('aws-sdk');
const params = require('./function-parameters.json');
exports.handler = async (event, context, callback) => {
  const cognitoidentityserviceprovider = new aws.CognitoIdentityServiceProvider({ apiVersion: '2016-04-18' });

  const groupParams = {
    GroupName: params.GROUP,
    UserPoolId: event.userPoolId,
  };

  const addUserParams = {
    GroupName: params.GROUP,
    UserPoolId: event.userPoolId,
    Username: event.userName,
  };

  try {
    await cognitoidentityserviceprovider.getGroup(groupParams).promise();
  } catch (e) {
    await cognitoidentityserviceprovider.createGroup(groupParams).promise();
  }

  try {
    await cognitoidentityserviceprovider.adminAddUserToGroup(addUserParams).promise();
    callback(null, event);
  } catch (e) {
    callback(e);
  }
};
```
and function-parameter.json contains
```
{
  "trigger": true,
  "modules": [
    "add-to-group"
  ],
  "parentResource": "auth",
  "functionName": "authPostConfirmation",
  "resourceName": "authPostConfirmation",
  "parentStack": "auth",
  "triggerEnvs": [],
  "triggerDir": "/home/deagle/.config/fnm/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/provider-utils/awscloudformation/triggers/PostConfirmation",
  "triggerTemplate": "PostConfirmation.json.ejs",
  "roleName": "authPostConfirmation",
  "skipEdit": true,
  "GROUP": "customers",
  "enableCors": false
}
```
the triggerDir seems to point to his local filesystem... could that be a problem?

**Comment by josefaidt** [comment]
> the triggerDir seems to point to his local filesystem... could that be a problem?

It does appear that it could be causing the issue. How was the project set up when you originally pulled it down? If the `amplify/.config/project-config.json` file has the appropriate path and this file was not updated with the correct path this may be a bug in the init process.

If you manually update that path does it alleviate the issue? Taking a look at the code where it's erroring out it may be due to an attempted write to a directory that does not exist, but perhaps the error can be surfaced in a more elegant manner.

**Comment by samlaf** [comment]
I tried changing the path to my computer's path, and I get the exact same error message and stack trace.

What do you mean by 'how was the project set up'? Do you mean originally when amplify was first setup, or when I cloned our repo to my computer and ran `amplify init`?

`amplify/.config/project-config.json` doesn't seem to contain any such path:
```
{
  "providers": [
    "awscloudformation"
  ],
  "projectName": "web-app",
  "version": "3.1",
  "frontend": "javascript",
  "javascript": {
    "framework": "vue",
    "config": {
      "SourceDir": "src",
      "DistributionDir": "dist",
      "BuildCommand": "yarn-script build",
      "StartCommand": "yarn-script serve"
    }
  }
}
```
I can't figure out what exactly is the `function-parameters.json` file that I showed you in my previous message. I used the default `amplify add auth` process in a separate project, and it generated:
![image](https://user-images.githubusercontent.com/9342524/135305984-45d37d04-1c1e-47a0-84fb-bf620b571335.png)
which we can compare to our directory:
![image](https://user-images.githubusercontent.com/9342524/135305895-3679e0af-6019-4e2d-8252-44ce06ca22d9.png)

Did we create this file? Or is this generated by amplify? If it is generated by amplify, is it supposed to be automatically changed to the local path when different developers pull from amplify?

**Comment by samlaf** [comment]
I just realized that I now get an error when I checkout a new env:
```
> amplify env checkout preprod
✔ Initialized provider successfully.
✖ There was an error initializing your environment.
Cannot read property 'replace' of undefined
TypeError: Cannot read property 'replace' of undefined
    at initTriggerEnvs (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/src/provider-utils/awscloudformation/index.ts:421:56)
    at Object.updateConfigOnEnvInit (/home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/src/provider-utils/awscloudformation/index.ts:374:25)
    at /home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-function/src/index.ts:131:28
    at /home/samlaf/.nvm/versions/node/v14.17.4/lib/node_modules/@aws-amplify/cli/node_modules/promise-sequential/index.js:16:18
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

Could this be related?

**Comment by josefaidt** [comment]
> What do you mean by 'how was the project set up'? Do you mean originally when amplify was first setup, or when I cloned our repo to my computer and ran amplify init?

Yes, `amplify init` should update the path values, which it sounds like you have already done.

> amplify/.config/project-config.json doesn't seem to contain any such path:

Ah, my apologies it is in the `amplify/.config/local-env-info.json` file.

> I can't figure out what exactly is the function-parameters.json file that I showed you in my previous message. I used the default amplify add auth process in a separate project, and it generated:

This file is likely hidden in VSCode due to the project settings added to `.vscode/settings.json`

```json
{
  "files.exclude": {
    "amplify/.config": true,
    "amplify/**/*-parameters.json": true,
    "amplify/**/amplify.state": true,
    "amplify/**/transform.conf.json": true,
    "amplify/#current-cloud-backend": true,
    "amplify/backend/amplify-meta.json": true,
    "amplify/backend/awscloudformation": true
  }
}
```

> Did we create this file? Or is this generated by amplify?

Yes it is created by Amplify when adding a Cognito trigger when stepping through `amplify add auth`

> If it is generated by amplify, is it supposed to be automatically changed to the local path when different developers pull from amplify?

Ideally, yes, it should be updated.

> I just realized that I now get an error when I checkout a new env:

This error is coming from where the CLI is looking for `resourceName` in `function-parameters.json`, which is quite odd considering in your comment that file has that property. I'm going to try pulling a project with this setup down in a separate directory and see if I can pinpoint the root cause. For what it's worth this is what my trigger function's `function-parameters.json` file looks like after initial creation on the latest version of Amplify CLI:

```json
{
  "trigger": true,
  "modules": [
    "add-to-group"
  ],
  "parentResource": "828026fefaa226fefaa2",
  "functionName": "828026fefaa226fefaa2PostConfirmation",
  "resourceName": "828026fefaa226fefaa2PostConfirmation",
  "parentStack": "auth",
  "triggerEnvs": [],
  "triggerDir": "/Users/josef/.config/yarn/global/node_modules/amplify-category-auth/provider-utils/awscloudformation/triggers/PostConfirmation",
  "triggerTemplate": "PostConfirmation.json.ejs",
  "triggerEventPath": "PostConfirmation.event.json",
  "roleName": "828026fefaa226fefaa2PostConfirmation",
  "skipEdit": true,
  "GROUP": "users",
  "enableCors": false
}
```

**Comment by samlaf** [comment]
My `local-env-info.json` does indeed contain the correct path on my computer:
```
{
  "projectPath": "/home/samlaf/dev/oa-media-cdk-stacks/web-app",
  "defaultEditor": "vscode",
  "envName": "staging"
}
```
As for `function-parameters.json`, it is hidden by my .vscode's config, but it still shows up in the explorer.
Anyhow, is it normal that there's a `function-parameters.json` inside the src directory, and another one outside?
![image](https://user-images.githubusercontent.com/9342524/135481797-e88aeef1-d25e-4398-b6c7-ec0ef4394018.png)
They are both slightly different from yours.
The one in src contains
```
{
  "trigger": true,
  "modules": [
    "add-to-group"
  ],
  "parentResource": "auth",
  "functionName": "authPostConfirmation",
  "resourceName": "authPostConfirmation",
  "parentStack": "auth",
  "triggerEnvs": [],
  "triggerDir": "/home/deagle/.config/fnm/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/provider-utils/awscloudformation/triggers/PostConfirmation",
  "triggerTemplate": "PostConfirmation.json.ejs",
  "roleName": "authPostConfirmation",
  "skipEdit": true,
  "GROUP": "customers",
  "enableCors": false
}
```
and the one outside contains
```
{
  "trigger": true,
  "modules": [
    "add-to-group"
  ],
  "parentResource": "auth",
  "functionName": "authPostConfirmation",
  "parentStack": "auth",
  "triggerEnvs": "[]",
  "triggerDir": "/home/deagle/.local/share/nvm/v14.16.0/lib/node_modules/@aws-amplify/cli/node_modules/amplify-category-auth/provider-utils/awscloudformation/triggers/PostConfirmation",
  "roleName": "authPostConfirmation",
  "triggerTemplate": "PostConfirmation.json.ejs",
  "triggerEventPath": "PostConfirmation.event.json",
  "skipEdit": true
}
```
and their diff:
![image](https://user-images.githubusercontent.com/9342524/135482196-4f74a5e3-5a52-47cd-b67d-877523f25aae.png)

**Comment by samlaf** [comment]
I just noticed that `local-env-info.json` is not checked-in to git, but the `function-parameters.json` are.
How are the `triggerDir`s supposed to get updated if they are checked into git? All of them (for the different functions) are set to my colleague's computer's local path.

**Comment by samlaf** [comment]
I added `resourceName: "authPostConfirmation"` to the function-parameter.json file outside of src and this fixed the bug!
It would be nice to have better error messages for such an error.

**Comment by josefaidt** [comment]
> is it normal that there's a function-parameters.json inside the src directory, and another one outside?

There should only be one of these files location in the function's directory itself, not its `src/` directory.

> I just noticed that local-env-info.json is not checked-in to git, but the function-parameters.json are.
How are the triggerDirs supposed to get updated if they are checked into git? All of them (for the different functions) are set to my colleague's computer's local path.

_Ideally_ this should be updated on `amplify init` when the project is initialized, however briefly looking at the code I don't believe this path is updated beyond initial creation. For that and the lack of helpful error messages I'll mark this as a bug 🙂 

https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-cli/src/extensions/amplify-helpers/trigger-flow.ts#L199

**Comment by jhockett** [comment]
@samlaf it looks like a PR was released some time ago to address this issue. Are you unblocked now?

**Comment by samlaf** [comment]
I changed company... so won't be able to verify this unfortunately.

**Comment by josefaidt** [comment]
Closing as issue is resolved
